pkg/updatestatus/nodeinformer_test.go

```diff
@@ -1051,3 +1105,140 @@ func Test_sync_with_node(t *testing.T) {
 		})
 	}
 }
+
+func Test_sync_with_mcp(t *testing.T) {
+	now := metav1.Now()
+
+	mcIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+	for _, o := range []metav1.Object{} {
+		if err := mcIndexer.Add(o); err != nil {
+			t.Fatalf("Failed to add o to indexer: %v", err)
+		}
+	}
+	mcLister := machineconfigv1listers.NewMachineConfigLister(mcIndexer)
+
+	nodeIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+	for _, o := range []metav1.Object{&corev1.Node{ObjectMeta: metav1.ObjectMeta{Name: "master-1"}},
+		&corev1.Node{ObjectMeta: metav1.ObjectMeta{Name: "worker-1"}}} {
+		if err := nodeIndexer.Add(o); err != nil {
+			t.Fatalf("Failed to add o to indexer: %v", err)
+		}
+	}
+	nodeLister := corelistersv1.NewNodeLister(nodeIndexer)
+
+	testCases := []struct {
+		name string
+
+		object                runtime.Object
+		pools                 []*machineconfigv1.MachineConfigPool
+		mcpToRemove, mcpToAdd string
+
+		expectedErr      error
+		expectedQueueLen int
+	}{
+		{
+			name: "reconcile for deleted pool",
+			object: &machineconfigv1.MachineConfigPool{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "non-exist-mcp",
+				},
+			},
+			pools:            []*machineconfigv1.MachineConfigPool{getMCP("master"), getMCP("worker"), getMCP("non-exist-mcp")},
+			mcpToRemove:      "non-exist-mcp",
+			expectedQueueLen: 2,
+		},
+		{
+			name: "reconcile for a new pool",
+			object: &machineconfigv1.MachineConfigPool{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "infra",
+				},
+			},
+			pools:    []*machineconfigv1.MachineConfigPool{getMCP("master"), getMCP("worker")},
+			mcpToAdd: "infra",
+		},
```

Comment:
Hmm, a reconcile is expected from my understanding but the `expectedQueueLen` is set to zero.

Reply:
Good catch. How come I did not notice it. Thanks.

Reply:
Done in https://github.com/openshift/cluster-version-operator/compare/791b701aba13b9760aeee99f6ca01e5e2f6a2967..4ff76c0109d0569ce6d4a20042c77d2be29cf5a9